### DISCUSSION TITLE
New XSharedPreferences means read only not support write in the hooked app, right?

### DISCUSSION THREAD

**Discussion by Jasper-1024** [LSPosed/LSPosed#1764] [Q&A] [answered]
My English is not very good, and I haven't done android development for a while, so I hope I can describe the problem clearly.

I am using New XSharedPreferences,tried to fix an old module,yes it work but XSharedPreferences is read only.
I try to use `MODE_WORLD_WRITEABLE`, not work.

then back to the wiki,as it sad:

---

Notice that LSPosed will help you to make the preference world readable if and only if you are getting preference in mode Context.MODE_WORLD_READABLE.

---

so it means New XSharedPreferences is read only not support write in the hooked app, right?

**Comment by yujincheng08** [comment, marked as answer]
yes

**Reply by Jasper-1024** [reply]
while thanks,then i have to find another way to let the hooked app could transfer information to the module.

**Reply by ghost** [reply]
I used this https://github.com/apsun/RemotePreferences

**Reply by Jasper-1024** [reply]
i has check this project,it seem useing ContentProvider.
actually i am useing ContentProvider directly now,beacuse ContentProvider not work on some version custom rom,i have to change to another way, thanks anyway.

**Reply by ghost** [reply]
yes it use a ContentProvider for read/write. It require a context in module but is easy to retrieve it.
I tested on LineageOs, MIUI and Samsung, I never seen an issue on ContentProvider and on RemotePreferences.
Good luck in your search, when you will find a solution,  if you want please share it.

**Reply by Jasper-1024** [reply]
> yes it use a ContentProvider for read/write. It require a context in module but is easy to retrieve it. I tested on LineageOs, MIUI and Samsung, I never seen an issue on ContentProvider and on RemotePreferences. Good luck in your search, when you will find a solution, if you want please share it.

perhaps i have been too brief in my description.
it is [nowakelock](https://github.com/Jasper-1024/NoWakeLock),another my project, always face a same problem,the hooked app could not transfer information to the module.
of course it is not because ContentProvider,ContentProvider work well,it just because using ContentProvider may cause the information about another app try to launch module app. this information is quite confuse let some user limit module app run on the background.

therefor i need find another way to avoid this information appear.
